clang/test/Sema/warn-lifetime-analysis-nocfg.cpp

```diff
@@ -601,17 +601,23 @@ void test() {
   std::optional<std::string_view> o4 = std::optional<std::string_view>(s); 
 
   // FIXME: should work for assignment cases
-  v1 = {std::string()};
-  o1 = std::string();
+  v1 = {std::string()}; // expected-warning {{object backing the pointer}}
+  o1 = std::string(); // expected-warning {{object backing the pointer}}
 
   // no warning on copying pointers.
   std::vector<std::string_view> n1 = {std::string_view()};
+  n1 = {std::string_view()};
```

Comment:
Not strictly related to this PR, but we could add some tests for more deeply nested containers like `std::vector<std::vector<std::string_view>>`. Regardless if we gat warnings for those or not, documenting whether this case works in tests have some value.

Reply:
Done, added. Current we only support the first-level nested container.